compiler/qsc_eval/src/backend.rs

```diff
@@ -295,6 +298,13 @@ impl Backend for SparseSim {
         self.apply_noise(q);
     }
 
+    fn sx(&mut self, q: usize) {
+        self.sim.h(q);
```

Comment:
having sx on sim would be preferred.

Reply:
That `sim` variable is the `QuantumSim` struct from [qir-runner](https://github.com/qir-alliance/qir-runner/blob/main/sparsesim/src/lib.rs#L35), which does not support SX natively. It could be extended to do so, but that would require changing how gate caching works because of the way SX and H would interact. That's why in https://github.com/qir-alliance/qir-runner/pull/221 the same technique was employed: expose the intrinsic but emulate it via H,S,H on the simulator.